server/src/routes/shift.js

```diff
@@ -0,0 +1,33 @@
+import express from "express";
+import { checkToken } from "../util/middleware.js";
+import * as validator from "../util/shiftMiddleware.js";
+import * as controller from "../controller/shiftController.js";
+import * as express_validator from "express-validator";
+
+const body = express_validator.body;
+const param = express_validator.param;
+
+const router = express.Router();
+
+router.post(
+  "/",
+  validator.isShiftId,
+  controller.createShift
+);
+
+router.get("/", controller.getShift);
+
+router.delete(
+  "/:id",
+  validator.isShiftId,
+  controller.deleteShift
+);
+
+router.post(
+  "/update",
+  validator.isShiftId,
```

Comment:
Also need to validate times and user exists here

Reply:
rewrite:

router.post(
  "/update",
  validator.isUserId, // make sure user exists
  validator.isValidTimeShift,
  validator.isShiftId, // make sure shift exists
  controller.updateShift
);